File: backend/app/schemas/location.py
from typing import Optional

from app.schemas.base import ORMModel
from pydantic import BaseModel


class LocationCreate(BaseModel):
    warehouse_id: int
    code: str
    zone_id: Optional[int] = None
    description: Optional[str] = None


class LocationUpdate(BaseModel):
    warehouse_id: Optional[int] = None
    zone_id: Optional[int] = None
    code: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None


class LocationRead(ORMModel):
    id: int
    warehouse_id: int
    zone_id: Optional[int] = None
    code: str
    description: Optional[str] = None
    is_active: bool

File: frontend/src/types/location.ts
export type Location = {
  id: number;
  warehouse_id: number;
  zone_id?: number | null;
  code: string;
  description?: string | null;
  is_active: boolean;
};

export type LocationCreate = {
  warehouse_id: number;
  code: string;
  zone_id?: number | null;
  description?: string | null;
};

export type LocationUpdate = Partial<LocationCreate> & {
  is_active?: boolean;
};
